client/python/armada_client/client.py

```diff
@@ -183,6 +183,46 @@ def get_queue_info(self, name: str):
         return response
 
 
+def search_for_job_complete(event, job_name: str, job_id: str) -> Tuple[str, str]:
```

Comment:
I understand why we have this but I'm not a massive fan of it.  This is potentially going to fetch millions of events from the server and filter them all out client side, just to read a single event.

I think long term we should make the lookout api into a public api that stores the state of the world and allows clients discover such facts with a much more efficient query.